src/mesido/esdl/esdl_heat_model.py

```diff
@@ -1600,6 +1637,221 @@ def convert_compressor(self, asset: Asset) -> Tuple[Type[Compressor], MODIFIERS]
 
         return Compressor, modifiers
 
+    def convert_gas_boiler(self, asset: Asset) -> Tuple[GasBoiler, MODIFIERS]:
+        """
+        This function converts the GasHeater object in esdl to a set of modifiers that can be
+        used in a pycml object.
+
+        Parameters
+        ----------
+        asset : The asset object with its properties.
+
+        Returns
+        -------
+        GasBoiler class with modifiers
+        """
+        assert asset.asset_type in {"GasHeater"}
+
+        max_supply = asset.attributes["power"]
+
+        if not max_supply:
+            logger.error(f"{asset.asset_type} '{asset.name}' has no max power specified. ")
+        assert max_supply > 0.0
+
+        if len(asset.in_ports) == 1:
+            heat_source_object, modifiers = self.convert_heat_source(asset)
+            return heat_source_object, modifiers
+
+        id_mapping = asset.global_properties["carriers"][asset.in_ports[0].carrier.id][
+            "id_number_mapping"
+        ]
+
+        for port in asset.in_ports:
+            if isinstance(port.carrier, esdl.GasCommodity):
+                density = self.get_density(asset.name, port.carrier)
+                internal_energy = self.get_internal_energy(asset.name, port.carrier)
+
+        co2_coefficient = 1.0
+        if hasattr(asset.attributes["KPIs"], "kpi"):
+            co2_coefficient = asset.attributes["KPIs"].kpi.items[0].value
```

Comment:
I'm not very happy with this. In case the user has added other KPIs they might here be used as CO2 cofficients because we only use the first item of the list.

Reply:
also agree, but I copied it from the convert source, do we delete it everywhere?

Reply:
damnn that I hadn't noticed. We might want to have CO2 info parsing, but that should be clearly defined. For now I would delte it here and comment it out at the source (add TODO) such that we remember to still include this.